### PR TITLE
feat: support pwsh in powershell provisioner

### LINKED ISSUE #7164: Support PowerShell 6+ as a provisioner
This is a feature request. This would reduce some headaches that have to be dealt with when using Windows PowerShell. 

https://docs.microsoft.com/en-us/powershell/scripting/learn/remoting/wsman-remoting-in-powershell-core?view=powershell-6

### PR DESCRIPTION
**Pull request by edeustace**
* set `pwsh` to true in config to run `pwsh.exe`, default is false.
* add docs

Closes #7164

### CODE REVIEW

**Review by lbajolet-hashicorp — APPROVED** [review]
Just a nit regarding a condition in the test, outside of that LGTM!

Thanks for the contribution, and sorry for the late review

File: provisioner/powershell/provisioner.go [hunk 1]
```diff
@@ -112,7 +115,12 @@ func (p *Provisioner) defaultExecuteCommand() string {
 		return baseCmd
 	}
 
-	return fmt.Sprintf(`powershell -executionpolicy %s "%s"`, p.config.ExecutionPolicy, baseCmd)
+	if(p.config.Pwsh){
+	  return fmt.Sprintf(`pwsh -executionpolicy %s -command "%s"`, p.config.ExecutionPolicy, baseCmd)
+	} else {
+	  return fmt.Sprintf(`powershell -executionpolicy %s "%s"`, p.config.ExecutionPolicy, baseCmd)
```
Comment (lbajolet-hashicorp) [inline provisioner/powershell/provisioner.go:121]: Re-reading through this PR, I notice the indentation is off here, could you replace the two-spaces by a tab? That'd help it stay consistent with the rest of the code and the Go guidelines.

File: provisioner/powershell/provisioner.go [hunk 2]
```diff
@@ -89,6 +89,9 @@ type Config struct {
 	// A duration of how long to pause after the provisioner
 	PauseAfter time.Duration `mapstructure:"pause_after"`
 
+	// Run pwsh.exe instead of powershell.exe - latest version of powershell.
+	Pwsh bool `mapstructure:"pwsh"`
```
Comment (lbajolet-hashicorp) [inline provisioner/powershell/provisioner.go:93]: As noted by @JenGoldstrich in a out-of-band discussion, `UsePwsh` would be a clearer name for the attribute, could you amend this?

**Review by lbajolet-hashicorp — CHANGES_REQUESTED** [review]
Coming back on this PR, after a re-reading of the code, we think it'd be beneficial to address a few remaining comments before we can merge this, any chance you can address those @edeustace?

**Comment by nywilken** [comment]
HI @edeustace thanks for the contribution. I'm going to close this PR as your changes, along with commit history, have been rolled into #11950. The team requested a few changes a bit late so we thought it best to make the changes along with yours in a separate PR. Thanks again for making the fix. 

We look forward to the continued contributions and feedback. Cheers!

**Comment by edeustace** [comment]
Sorry - I was on vacation - thanks!

**Pull request closed without merging**